### ISSUE TITLE
Resource timing entries are never added to the performance entry buffer

### ISSUE THREAD

**Issue by jdm** [servo/servo#23328]
NetworkListener::submit_timing has this code:
```
    global
        .performance()
        .queue_entry(performance_entry.upcast::<PerformanceEntry>(), false);
```
The `false` means "do not add this performance entry to the performance entry buffer". This means that any JS code that uses `performance.getEntries`, or `performance.getEntriesByType` or `performance.getEntriesByName` will never observe this resource timing information. I filed https://github.com/w3c/performance-timeline/issues/128 to clarify when this flag is actually supposed to be true; in the short term it might make sense to pass true instead.

**Comment by gatoWololo** [comment]
Hello,

I'm currently looking at this issue. I have reviewed the relevant links in https://github.com/w3c/performance-timeline/issues/128 but it seems that so far there isn't a concrete answer? So I'm not sure exactly what should be done. Just change `false` -> `true`? Or do we need to go through the spec and figure out exactly when the flag should be set?

**Comment by jdm** [comment]
Yes, we should go ahead and change the false to true to ensure we get entries when we expect. You can run the tests at tests/wpt/web-platform-tests/resource-timing/ to see if there are any changes to the test results.

**Comment by gatoWololo** [comment]
I'm seeing a lot of errors. More than I expected and for more varied that I expected. Thoughts?
```
  ▶ Unexpected subtest result in /resource-timing/test_resource_timing.https.html:
  └ PASS [expected FAIL] window.performance.getEntriesByName() and window.performance.getEntriesByNameType() return same data (img)

  ▶ Unexpected subtest result in /resource-timing/test_resource_timing.html:
  └ PASS [expected FAIL] window.performance.getEntriesByName() and window.performance.getEntriesByNameType() return same data (img)

  ▶ Unexpected subtest result in /resource-timing/resource_dedicated_worker.html:
  └ PASS [expected FAIL] There should be six entries: 4 scripts, 1 stylesheet, and the worker itself

  ▶ Unexpected subtest result in /resource-timing/no-entries-for-cross-origin-css-fetched.sub.html:
  │ FAIL [expected PASS] Make sure that resources fetched by cross origin CSS are not in the timeline.
  │   → assert_equals: Import should not be in timeline expected 0 but got 1

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_zero.htm:
  └ PASS [expected FAIL] fetchStart should be greater than 0 in cross-origin request.

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_zero.htm:
  │ FAIL [expected PASS] There should be one resource timing entry.
  │   → assert_equals: There should be one resource timing entry. expected 1 but got 2

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_zero.htm:
  │ FAIL [expected PASS] connectStart should be 0 in cross-origin request.
  │   → assert_equals: connectStart should be 0 in cross-origin request. expected 0 but got 33494162

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_zero.htm:
  │ FAIL [expected PASS] connectEnd should be 0 in cross-origin request.
  │   → assert_equals: connectEnd should be 0 in cross-origin request. expected 0 but got 33494162

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_zero.htm:
  │ FAIL [expected PASS] requestStart should be 0 in cross-origin request.
  │   → assert_equals: requestStart should be 0 in cross-origin request. expected 0 but got 33494162823884

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.html:
  │ FAIL [expected PASS] domainLookupEnd and fetchStart should be the same
  │   → assert_equals: domainLookupEnd and fetchStart should be the same expected (undefined) undefined but got (number) 33494634127758

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.html:
  │ FAIL [expected PASS] connectStart and fetchStart should be the same
  │   → assert_equals: connectStart and fetchStart should be the same expected 33494634 but got 33494634127758

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.html:
  │ FAIL [expected PASS] connectEnd and fetchStart should be the same
  │   → assert_equals: connectEnd and fetchStart should be the same expected 33494634 but got 33494634127758

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.html:
  │ FAIL [expected PASS] secureConnectionStart should be zero
  │   → assert_equals: secureConnectionStart should be zero expected (number) 0 but got (undefined) undefined

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.html:
  │ FAIL [expected PASS] domainLookupStart and fetchStart should be the same
  │   → assert_equals: domainLookupStart and fetchStart should be the same expected (undefined) undefined but got (number) 33494634127758

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming must be primary interface of resource

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] Stringification of resource

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "initiatorType" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "nextHopProtocol" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "redirectStart" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "fetchStart" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "connectStart" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "connectEnd" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "requestStart" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "responseStart" with the proper type

  ▶ Unexpected subtest result in /resource-timing/idlharness.any.worker.html:
  └ PASS [expected FAIL] PerformanceResourceTiming interface: resource must inherit property "responseEnd" with the proper type

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content_redirect.html:
  │ FAIL [expected PASS] domainLookupEnd and fetchStart should be the same
  │   → assert_equals: domainLookupEnd and fetchStart should be the same expected (undefined) undefined but got (number) 33506430457074

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content_redirect.html:
  │ FAIL [expected PASS] connectEnd and fetchStart should be the same
  │   → assert_equals: connectEnd and fetchStart should be the same expected 33506430 but got 33506430457074

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content_redirect.html:
  │ FAIL [expected PASS] domainLookupStart and fetchStart should be the same
  │   → assert_equals: domainLookupStart and fetchStart should be the same expected (undefined) undefined but got (number) 33506430457074

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content_redirect.html:
  │ FAIL [expected PASS] secureConnectionStart and fetchStart should be the same
  │   → assert_equals: secureConnectionStart and fetchStart should be the same expected (undefined) undefined but got (number) 33506430457074

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content_redirect.html:
  │ FAIL [expected PASS] connectStart and fetchStart should be the same
  │   → assert_equals: connectStart and fetchStart should be the same expected 33506430 but got 33506430457074

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.https.html:
  │ FAIL [expected PASS] connectStart and fetchStart should be the same
  │   → assert_equals: connectStart and fetchStart should be the same expected 33506511 but got 33506511845506

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.https.html:
  │ FAIL [expected PASS] secureConnectionStart and fetchStart should be the same
  │   → assert_equals: secureConnectionStart and fetchStart should be the same expected (undefined) undefined but got (number) 33506511845506

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.https.html:
  │ FAIL [expected PASS] connectEnd and fetchStart should be the same
  │   → assert_equals: connectEnd and fetchStart should be the same expected 33506511 but got 33506511845506

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.https.html:
  │ FAIL [expected PASS] domainLookupEnd and fetchStart should be the same
  │   → assert_equals: domainLookupEnd and fetchStart should be the same expected (undefined) undefined but got (number) 33506511845506

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse.https.html:
  │ FAIL [expected PASS] domainLookupStart and fetchStart should be the same
  │   → assert_equals: domainLookupStart and fetchStart should be the same expected (undefined) undefined but got (number) 33506511845506

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_origin.htm:
  └ PASS [expected FAIL] connectStart should not be 0 in timing-allow cross-origin request.

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_origin.htm:
  └ PASS [expected FAIL] connectEnd should not be 0 in timing-allow cross-origin request.

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_origin.htm:
  └ PASS [expected FAIL] requestStart should not be 0 in timing-allow cross-origin request.

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_origin.htm:
  └ PASS [expected FAIL] fetchStart should not be 0 in timing-allow cross-origin request.

  ▶ Unexpected subtest result in /resource-timing/resource_TAO_origin.htm:
  │ FAIL [expected PASS] The iframe should have one resource timing entry.
  │   → assert_equals: The iframe should have one resource timing entry. expected 1 but got 2

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content.html:
  │ FAIL [expected PASS] domainLookupStart and fetchStart should be the same
  │   → assert_equals: domainLookupStart and fetchStart should be the same expected (undefined) undefined but got (number) 33507703371627

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content.html:
  │ FAIL [expected PASS] connectEnd and fetchStart should be the same
  │   → assert_equals: connectEnd and fetchStart should be the same expected 33507703 but got 33507703371627

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content.html:
  │ FAIL [expected PASS] connectStart and fetchStart should be the same
  │   → assert_equals: connectStart and fetchStart should be the same expected 33507703 but got 33507703371627

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content.html:
  │ FAIL [expected PASS] secureConnectionStart and fetchStart should be the same
  │   → assert_equals: secureConnectionStart and fetchStart should be the same expected (undefined) undefined but got (number) 33507703371627

  ▶ Unexpected subtest result in /resource-timing/resource_connection_reuse_mixed_content.html:
  │ FAIL [expected PASS] domainLookupEnd and fetchStart should be the same
  │   → assert_equals: domainLookupEnd and fetchStart should be the same expected (undefined) undefined but got (number) 33507703371627
```

**Comment by jdm** [comment]
Hey, great! That means that a bunch of those tests were relying on resource timing entries that we weren't generating, so we were only "passing" them by accident.

**Comment by jdm** [comment]
A bunch of these failures look like they're related to our actual implementation of resource timing and cross-origin checks, so it's good that this change exposes the fact that we're missing some checks and properties.

**Comment by gatoWololo** [comment]
Ah, I see. So what should the next steps be?

**Comment by jdm** [comment]
Go ahead and update the expected test results (you can use https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations to make it easy) and make a pull request with your changes.

**Comment by gatoWololo** [comment]
Should I be updating _all_ the wpt tests. Or only those in `./tests/wpt/web-platform-tests/resource-timing/`?

I tried just doing:
`./mach test --log-raw /tmp/servo_timing.log --release ./tests/wpt/web-platform-tests/resource-timing/` but it errors with `--log-raw is not a valid test path or suite name`

Instead I tried: `/mach test --release ./tests/wpt/web-platform-tests/resource-timing/ --log-raw /tmp/servo_timing.log` but no long is produced (I'm guessing this is not parsing like I think it should be)


Doing a `./mach test-wpt --release --log-raw /tmp/servo_timing.log` seems pretty broad, specially since I'm seeing messages unrelated to the resource timing change:
```
  ▶ PASS [expected FAIL] /css/CSS2/floats/floats-in-table-caption-001.html

  ▶ FAIL [expected PASS] /css/CSS2/text/white-space-002.xht
  └   → /css/CSS2/text/white-space-002.xht 10a33a5df366a8b76dd0427be06b4e26ec361d27
/css/CSS2/text/white-space-002-ref.xht 4078c2610bbcded6f85648ddfa7a704d00790130

  ▶ FAIL [expected PASS] /css/CSS2/text/white-space-processing-056.xht
  └   → /css/CSS2/text/white-space-processing-056.xht 3de5e7b219b5b94324301554e9e75ed0f7c6d393
/css/CSS2/text/white-space-processing-056-ref.xht 63abe593f7ffb8aa65a9512a64b05384df80a6bd
...
```
So I worry about `update-wpt`-ing more tests than I should

**Comment by jdm** [comment]
You can do `./mach test-wpt --release --log-raw /tmp/servo_timing.log tests/wpt/web-platform-tests/resource-timing/` and that will do what you want.

**Comment by jdm** [comment]
This was fixed by #23536
